File: metal-drawing-a-triangle-with-metal-4/View controller/ViewController.m
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
The implementation for the cross-platform view controller.
*/

#import "ViewController.h"
#import "MetalKitViewDelegate.h"

@implementation ViewController
{
    MTKView *view;
    MetalKitViewDelegate *delegate;
}

- (void) viewDidLoad
{
    // Call the super class's method first.
    [super viewDidLoad];

    // Store a reference to the app's main view.
    view = (MTKView *)self.view;

    // Set the view's device property to the system's default Metal device.
    view.device = MTLCreateSystemDefaultDevice();
    NSAssert(view.device, @"Metal doesn't support this device.");

    delegate = [[MetalKitViewDelegate alloc] initWithMetalKitView:view];
    NSAssert(delegate, @"The view controller can't make a delegate for the MetalKit view.");

    view.delegate = delegate;
}

@end

File: foundation-icloud-synchronizing_app_preferences_with_icloud/PrefsInCloud_iOS/ViewController.swift
/*
See LICENSE folder for this sample’s licensing information.

Abstract:
The main view controller for this app.
*/

import UIKit

class ViewController: UIViewController {
	
    override func viewDidLoad() {
        super.viewDidLoad()
		
        // Default color background locally is "white".
        UserDefaults.standard.register(defaults: [gBackgroundColorKey: ColorIndex.white.rawValue])

        // Make sure we're showing the latest color.
        updateUserInterface()
		
        // Setup out key value store.
        prepareKeyValueStoreForUse()
    }

    // MARK: - Updates
    
    func updateUserInterface() {
        if let valideColorIndex = ColorIndex(rawValue: chosenColorValue) {
            view.backgroundColor = valideColorIndex.color
        }
    }

    // MARK: - Segue Support
    
    @IBAction func unwindToMenu(segue: UIStoryboardSegue) {
        if let colorsTableViewController = segue.source as? ColorsTableViewController,
            let row = colorsTableViewController.selectedIndexPath?.row, ColorIndex(rawValue: row) != nil {
            chosenColorValue = row
        }
    }
	
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Pass on the available background colors to the table view
        guard let navController = segue.destination as? UINavigationController else { return }
        guard let destinationVC = navController.topViewController as? ColorsTableViewController else { return }
        
        destinationVC.colorStrings = [ColorIndex.white.name, ColorIndex.red.name,
                                      ColorIndex.green.name, ColorIndex.yellow.name]
        destinationVC.selectedIndexPath = IndexPath(row: chosenColorValue, section: 0)
    }
	
}

